vim-patch: update runtime files

clason opened the pull request:
###  vim-patch:3afc9f2556fa

runtime(cmake): sync runtime files with upstream (vim/vim#13597)

https://github.com/vim/vim/commit/3afc9f2556faf2574d7950d879f2bf61612f3367

Co-authored-by: Eisuke Kawashima <e-kwsm@users.noreply.github.com>

### vim-patch:1c97b5c0c0b4

runtime(vim): Update syntax file, fix missing for highlight (vim/vim#13668)

Fix highlighting of :for command.  Link the vimFor syntax group to the
vimCommand highlight group.

Error introduced in commit f686921

https://github.com/vim/vim/commit/1c97b5c0c0b4a60d652b78da4470aeb8a7509700

Co-authored-by: dkearns <dougkearns@gmail.com>

### vim-patch:fa920da283f6

omit (filetype detection never depends on `fileignorecase` option in Neovim)

### vim-patch:ca7c9b1b5926,34b9a15c378f

omit (json formating plugin in vim9script)

### vim-patch:b52e7ac7c61c

runtime(netrw): Fix `w:netrw_bannercnt` ref error with `netrw_fastbrowse=2` (vim/vim#13659)

* NetRW: Fix `w:netrw_bannercnt` ref error with `netrw_fastbrowse=2`
* NetRW: Fix wrong `w:netrw_bannercnt` setting

closes: vim/vim#13640

https://github.com/vim/vim/commit/b52e7ac7c61cd88d04ee6b7b1808c361dd4a7640

Co-authored-by: KSR-Yasuda <31273423+KSR-Yasuda@users.noreply.github.com>

### vim-patch:5a68cdf14915

runtime(sbt): do not set b:did_ftplugin before sourcing scala ftplugin(vim/vim#13657)

The `b:did_ftplugin` guard was set and prevented us from actually sourcing `ftplugin/scala.vim`. Since the latter script also sets the guard properly, we can simply remove the guard here.

https://github.com/vim/vim/commit/5a68cdf14915f20a1a6889cd0a76d29ecd374f24

Co-authored-by: Karl Yngve Lervåg <karl.yngve@lervag.net>

### vim-patch:f6869212c9e1

runtime(vim): Update syntax file (vim/vim#13653)

Improve variable highlighting in :let, :unlet, :const and :for commands.

Match registers and local, global and terminal option variables.

https://github.com/vim/vim/commit/f6869212c9e19ec8f2a4434732dd28f84f170163

Co-authored-by: dkearns <dougkearns@gmail.com>

clason committed:
vim-patch:f6869212c9e1

runtime(vim): Update syntax file (vim/vim#13653)

Improve variable highlighting in :let, :unlet, :const and :for commands.

Match registers and local, global and terminal option variables.

https://github.com/vim/vim/commit/f6869212c9e19ec8f2a4434732dd28f84f170163

Co-authored-by: dkearns <dougkearns@gmail.com>

seandewar commented on runtime/syntax/vim.vim:
```diff
@@ -368,8 +371,19 @@ syn match	vimSetMod	contained	"&vim\=\|[!&?<]\|all&"
 " Let: {{{2
 " ===
 syn keyword	vimLet	let	unl[et]	skipwhite nextgroup=vimVar,vimFuncVar,vimLetHereDoc
```
```suggestion
```

clason committed:
vim-patch:5a68cdf14915

runtime(sbt): do not set b:did_ftplugin before sourcing scala ftplugin(vim/vim#13657)

The `b:did_ftplugin` guard was set and prevented us from actually sourcing `ftplugin/scala.vim`. Since the latter script also sets the guard properly, we can simply remove the guard here.

https://github.com/vim/vim/commit/5a68cdf14915f20a1a6889cd0a76d29ecd374f24

Co-authored-by: Karl Yngve Lervåg <karl.yngve@lervag.net>

clason committed:
vim-patch:b52e7ac7c61c

runtime(netrw): Fix `w:netrw_bannercnt` ref error with `netrw_fastbrowse=2` (vim/vim#13659)

* NetRW: Fix `w:netrw_bannercnt` ref error with `netrw_fastbrowse=2`
* NetRW: Fix wrong `w:netrw_bannercnt` setting

closes: vim/vim#13640

https://github.com/vim/vim/commit/b52e7ac7c61cd88d04ee6b7b1808c361dd4a7640

Co-authored-by: KSR-Yasuda <31273423+KSR-Yasuda@users.noreply.github.com>

seandewar commented on runtime/syntax/vim.vim:
```diff
@@ -368,8 +371,19 @@ syn match	vimSetMod	contained	"&vim\=\|[!&?<]\|all&"
 
+" For: {{{2
+" ===
+syn keyword	vimFor	for	skipwhite nextgroup=vimVar,vimVarList
```
Not your fault, but probably missing a `:hi def link vimFor vimCommand`

clason replied:
Will wait a bit then for a possible fixup.

clason replied:
They fixed it 🤗

seandewar reviewed (approved):
Others LGTM too. I've raised the `vimFor` issue over at Vim HQ.

zeertzjq commented on runtime/lua/vim/filetype/detect.lua:
```diff
@@ -245,6 +245,13 @@ function M.cpp(_, _)
 
+--- @type vim.filetype.mapfn
+function M.CH_fic(_, _)
+  if not vim.o.fileignorecase then
```
I don't think this is applicable. Unlike Vim, Nvim never respects 'fileignorecase' when matching filetype, because of Lua.

clason replied:
Do you mean "it _shouldn't_ be applicable? Because you can absolutely make it work if you want to.

zeertzjq replied:
The problem the patch was trying to solve is that `.h` files are still recognized as `cpp` when `g:c:syntax_for_h` and 'fileignorecase' is set.

Without this patch:
```sh
vim --clean --cmd 'let g:c_syntax_for_h = 1 | set fileignorecase' file.h
```
set `cpp` filetype, while
```sh
nvim --clean --cmd 'let g:c_syntax_for_h = 1 | set fileignorecase' file.h
```
already sets `c` filetype.

zeertzjq replied:
And, with this patch and 'fileignorecase' but without `g:c_syntax_for_h`, Vim set `c` filetype for `.C` files and `cpp` filetype for `.H` files, while Nvim set no filetype at all for `.C` and `.H` files.

clason replied:
OK, I'll leave this one for you to handle then.

zeertzjq replied:
TBH I don't know what to do either...

clason committed:
vim-patch:1c97b5c0c0b4

runtime(vim): Update syntax file, fix missing for highlight (vim/vim#13668)

Fix highlighting of :for command.  Link the vimFor syntax group to the
vimCommand highlight group.

Error introduced in commit f686921

https://github.com/vim/vim/commit/1c97b5c0c0b4a60d652b78da4470aeb8a7509700

Co-authored-by: dkearns <dougkearns@gmail.com>

clason committed:
vim-patch:3afc9f2556fa

runtime(cmake): sync runtime files with upstream (vim/vim#13597)

https://github.com/vim/vim/commit/3afc9f2556faf2574d7950d879f2bf61612f3367

Co-authored-by: Eisuke Kawashima <e-kwsm@users.noreply.github.com>
